packages/shared_preferences/shared_preferences_web/lib/shared_preferences_web.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show json;
+import 'dart:html' as html;
+
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
+
+/// The web implementation of [SharedPreferencesStorePlatform].
+///
+/// This class implements the `package:shared_preferences` functionality for the web.
+class SharedPreferencesPlugin extends SharedPreferencesStorePlatform {
+  /// Registers this class as the default instance of [SharedPreferencesStorePlatform].
+  static void registerWith(Registrar registrar) {
+    SharedPreferencesStorePlatform.instance = SharedPreferencesPlugin();
+  }
+
+  @override
+  Future<bool> clear() async {
+    // IMPORTANT: Do not use html.window.localStorage.clear() as that will
+    //            remove _all_ local data, not just the keys prefixed with
+    //            "flutter."
+    for (String key in _storedFlutterKeys) {
+      html.window.localStorage.remove(key);
+    }
+    return true;
+  }
+
+  @override
+  Future<Map<String, Object>> getAll() async {
+    final Map<String, Object> allData = <String, Object>{};
+    for (String key in _storedFlutterKeys) {
+      allData[key] = _decodeValue(html.window.localStorage[key]);
+    }
+    return allData;
+  }
+
+  @override
+  Future<bool> remove(String key) async {
+    _checkPrefix(key);
+    html.window.localStorage.remove(key);
+    return true;
+  }
+
+  @override
+  Future<bool> setValue(String valueType, String key, Object value) async {
+    _checkPrefix(key);
+    html.window.localStorage[key] = _encodeValue(value);
```

Comment:
Does it make sense to prepend something to the user-supplied `key`, to minimize collisions with other things that might be using the shared local storage? `shared_preferences` or similar?

We can make this transparent on all add/remove operations so users don't even see that we're doing this (unless they go to the dev tools and peek :P)

Reply:
The `SharedPreferences` class from `package:shared_preferences` already prepends "flutter.". So here I'm just verifying that it's prefixed in `_checkPrefix`. On top of that, for security reasons, `localStorage` is scoped to the domain, so you can't step on another site's data.

We could allow customizing the prefix, but currently the value is hard-coded across Dart, Java, and Objective-C code, so it would be a much bigger change.

Reply:
Ah ok, I thought this relied on the user prepending "flutter." to the key themselves. If it's the shared_prefs plugin, I guess it's OK.

I know you can't step on other's site data; the scenario I was envisioning was multiple plugins writing to localStorage on the same site, and overwriting each other's data.